src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -930,7 +930,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     var lazySemanticModel = body == null ? null : new Lazy<SemanticModel>(() =>
                     {
                         var syntax = body.Syntax;
-                        var semanticModel = (CSharpSemanticModel)_compilation.GetSemanticModel(syntax.SyntaxTree);
+                        var semanticModel = (CSharpSemanticModel)AnalyzerDriver.GetOrCreateCachedSemanticModel(syntax.SyntaxTree, _compilation, _cancellationToken);
```

Comment:
@CyrusNajmabadi I think the change here is probably going to be bigger then initially anticipated, and you should file a bug about this and assign it to me. I am still keen on investigating why the allocations/perf did not improve after your change, I'll do so later today. Thanks for your help here.